src/main/java/yoda/Yoda.java

```diff
@@ -0,0 +1,131 @@
+package yoda;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the chatbot.
+ * @author Tonishka Singh
+ */
+public class Yoda {
+    private Ui ui;
+    private Storage storage;
+    private QuestList questList;
+
+    public Yoda() {
+        this.ui = new Ui();
+        this.storage = new Storage();
+        this.questList = new QuestList(this.storage.load());
+    }
+
+    public static void main(String[] args) {
+        new Yoda().run();
+    }
+
+     private void run() {
+        ui.welcome();
+        Scanner sc = new Scanner(System.in);
+        Parser parser = new Parser();
+        while (true) {
+            System.out.print(">> ");
+            String input = sc.nextLine();
+            String[] args = parser.clean(input);
+            String command = parser.getCommand(args);
+
+            if (command.equals("bye")) {
+                sc.close();
+                ui.bye(storage, questList);
+                break;
+            } else if (command.equals("log")) {
+                log();
+            } else if (command.equals("mark")) {
+                int questID = parser.parseMark(args);
+                markDone(questID);
+                storage.save(questList);
+            } else if (command.equals("unmark")) {
+                int questID = parser.parseMark(args);
+                markUndone(questID);
+                storage.save(questList);
+            } else if (command.equals("todo")){
+                String desc = parser.getDescriptionToDo(args);
+                todo(desc);
+                storage.save(questList);
+            } else if (command.equals("event")) {
+                String[] argsEvent = parser.parseEvent(input);
+                event(argsEvent);
+                storage.save(questList);
+            } else if (command.equals("deadline")) {
+                String[] argsDeadline = parser.parseDeadline(input);
+                deadline(argsDeadline);
+                storage.save(questList);
+            } else if (command.equals("delete")) {
+                int questID = parser.parseMark(args);
+                delete(questID);
+                storage.save(questList);
+            } else if (command.equals("help")) {
+                ui.help();
+            } else if (command.equals("find")) {
+                find(parser.parseFind(args));
+            } else {
+                System.out.println("yoda.Yoda knows not what this means.");
+            }
+        }
+    }
+
+    private void log() {
+        System.out.println("To defeat the dark lord, " +
+                "following quests must you finish, Jedi:");
+        System.out.print(questList.toString());
+    }
+
+    private void todo(String desc) {
```

Comment:
Perhaps this could be changed to createTodo or makeTodo in order to make sure its phrased as a verb. The same applies for the following methods.